common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java

```diff
@@ -466,14 +468,14 @@ public void onReceivedClientCertRequest(WebView view,
     }
 
     /**
-     * A wrapper to stop YubiKitManager instance from detecting any more Usb devices.
+     * A wrapper to stop a smartcard manager instance from detecting any more Usb devices.
      */
-    public void stopYubiKitManagerUsbDiscovery() {
-        final String methodTag = TAG + ":stopYubiKitManagerUsbDiscovery";
+    public void stopSmartcardUsbDiscovery() {
+        final String methodTag = TAG + ":stopSmartcardUsbDiscovery";
         if (mClientCertAuthChallengeHandler != null) {
-            mClientCertAuthChallengeHandler.stopYubiKitManagerUsbDiscovery();
+            mClientCertAuthChallengeHandler.stopSmartcardUsbDiscovery();
```

Comment:
I'd suggest adding onDestroy() to the IChallengeHandler<> interface, and then invoke stopSmartCardUspDiscovery() there.

That way, the implementation detail won't be leaked.

Reply:
(and in the outer layer, WebviewAuthorizationFragment.onDestroy() can invoke IChallengeHandler.onDestroy() directly.

Reply:
Since `WebviewAuthorizationFragment` holds `AADWebViewClient` and not `IChallengeHandler`, I have it call `AADWebViewClient.onDestroy()` (I didn't add `onDestroy()` to the WebClient interface, since there are so many, but let me know if it would be necessary).